### PR TITLE
Gripper high level

### PR DESCRIPTION
**Pull request by GaiaZano05**
Its not the Gripper high level PR, it's the friends we made along the way

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+public class Gripper extends SubsystemBase {
+    private static Gripper INSTANCE;
+    private final GripperInputs inputs = new GripperInputs();
+    private final GripperIO io;
+    private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
+    private final MechanismRoot2d root = mechanism2d.getRoot("Gripper", 0, 0);
+    private final MechanismLigament2d shoulder =
+            root.append(new MechanismLigament2d("Gripper", 0, 0));
+    private final GripperInputsAutoLogged input = new GripperInputsAutoLogged();
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Like in the elevator, this should be made and taken from the `GripperIO`.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java:1]: Missing commands.

File: src/main/java/frc/robot/subsystems/gripper/GripperInputs.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package frc.robot.subsystems.gripper;
+
+import edu.wpi.first.units.*;
+import org.littletonrobotics.junction.AutoLog;
+
+@AutoLog
+public class GripperInputs {
+    public MutableMeasure<Angle> currentAngle;
+    public MutableMeasure<Angle> angleSetpoint;
+    public double rollerSpeed;
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/GripperInputs.java]: Is this the setpoint for the roller?

**Review by vichik123 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package frc.robot.subsystems.gripper;
+
+import edu.wpi.first.units.*;
+
+public interface GripperIO {
+
+    GripperInputs inputs = new GripperInputs();
+
+    void setSpeedMotorPower(double power);
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: Maybe choose a better name for this motor, like roller motor.

**Review by vichik123 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public GripperIO.ControlMode controlMode;
+
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
+
+    private final MechanismRoot2d root = mechanism2d.getRoot("Gripper", 0, 0);
+    private final MechanismLigament2d gripper =
+            root.append(new MechanismLigament2d("Gripper", 0, 0));
+    private final GripperInputsAutoLogged inputs = new GripperInputsAutoLogged();
+
+    public Gripper(GripperIO io) {
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Should be private. Remember to add `getInstance` and `initialize`.

**Review by vichik123 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/GripperInputs.java (hunk 2 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/gripper/GripperInputs.java]: The `rollerSpeed` I mean.

Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperInputs.java:10]: no, for the angle motor

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java (hunk 3 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: 84fa821bbcf5ff97709d179680bd8507dd11cb08

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 1 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 8cb9a19153fb50cc13351f1d45f0860b4e10e42c

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 4 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 818fcb731f6616991993e49c9c0811adfd1c67ee

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java:1]: 005f67da9b18bd630d06f32cbd79492075479676

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java (hunk 3 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: I meant for you to change it everywhere.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java:1]: 005f67da9b18bd630d06f32cbd79492075479676

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 4 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 818fcb731f6616991993e49c9c0811adfd1c67ee

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 1 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 8cb9a19153fb50cc13351f1d45f0860b4e10e42c

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java (hunk 3 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: fixed it

File: src/main/java/frc/robot/subsystems/gripper/GripperInputs.java (hunk 2 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperInputs.java]: yes, change the name?

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    public Command setRollerPower(double power){
+       return run(()-> setRollerMotorPower(power));
+    }
+    public Command intake(MutableMeasure<Angle> angle, double power){
+       return run(()-> setWristPosition(angle).alongWith(setRollerPower(power)));
+    }
+    public Command outtake(MutableMeasure<Angle> angle, double power){
+        return run(()-> setWristPosition(angle).alongWith(setRollerPower(power)));
+    }
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Is there a reason to have `void` setters that are used in `Command` methods? Why not just put everything in the command method? This should be changed in the elevator too BTW.

Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: IMO the whole reason for having "intake" and "outtake" methods is for convenience, so they shouldn't receive parameters and use constant values instead. Otherwise, the only difference between `intake` and `outtake` is the name...

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    public Command setWristPosition(MutableMeasure<Angle> angle){;
+        return run(()->this.setAngle(angle));
+    }
+
+    @Override
+    public void periodic() {
+        io.updateInputs(inputs);
+        Logger.processInputs(this.getClass().getSimpleName(), inputs);
+
+        Logger.recordOutput("current angle", getCurrentAngle());
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Why is that needed?

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+public class Gripper extends SubsystemBase {
+    private static Gripper INSTANCE;
+    private final GripperIO io;
+    public GripperIO.ControlMode controlMode;
+    private final GripperInputsAutoLogged inputs = GripperIO.inputs;
+
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
+
+    private final MechanismRoot2d root = mechanism2d.getRoot("Gripper", 0, 0);
+    private final MechanismLigament2d gripper =
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 1. "gripper" is a bit general name.
2. It's not being updated anywhere.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 7 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: What other name do you suggest?

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 5 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 588de2699f921294cbf6d4fb13b0610d8a6d491a

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 6 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: f6fafa5931444c17e23a9565262df1f539a5477f

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 7 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: f6fafa5931444c17e23a9565262df1f539a5477f

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 5 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 268aee32d232017edc9f886d332ee732764e2229

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java (hunk 3 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: 6cdb54b410aaadc666886503a5a5ae5e509ef41c

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 5 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Nice, but you can delete the `void` methods now. Also, should probably use `runOnce()`.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 7 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: This is not a relevant commit.
> What other name do you suggest?

A more specific name can be simply `gripperLigament`. The reasoning is that "gripper" can mean many things in a subsystem with the same name, so you'd want to be more clear in your naming.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 5 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: This will not work as expected. For example in the new `outtake()` method, `() -> setWristPosition(GripperConstants.OUTTAKE_ANGLE)` will not do anything. Remember that now, `setWristPosition` returns a command, but doesn't run. You need to slightly modify the new command methods for them to work.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+                .alongWith(run(()->io.setRollerMotorPower(GripperConstants.INTAKE_POWER)));
+    }
+    public Command outtake() {
+        return run(() -> io.setAngle(GripperConstants.OUTTAKE_ANGLE))
+                .alongWith(run(()->io.setRollerMotorPower(GripperConstants.OUTTAKE_POWER)));
+    }
+
+    public Command setWristPosition(MutableMeasure<Angle> angle) {
+        return run(() -> io.setAngle(angle));
+    }
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Forgot to comment that on the elevator too -- you need to use the `withName()` decorator. Otherwise, the name of the command will be e.g. `ParallelCommandGroup`.

**Review by katzuv — CHANGES_REQUESTED** [review]

**Review by katzuv — CHANGES_REQUESTED** [review]
Missing [`Command.withName()`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Command.html#withName(java.lang.String)) decorators everywhere (should be written like normal text and not in `camelCase`).

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 5 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: a9a7f8dfadb7957a07ddde7c35eb08e01e745718

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 8 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: d02a228ce282deb1e543cab9de7e271a218c697f

Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 1943b2706203c8a576bced6c58b04d25464d360b

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 7 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: d02a228ce282deb1e543cab9de7e271a218c697f

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 5 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 2dce6f79d41f638fb61c331733802a510e43ed27

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    public MutableMeasure<Voltage> getRollerMotorVoltage() {
+        return inputs.rollerMotorVoltage;
+    }
+
+    public MutableMeasure<Voltage> getAngleMotorVoltage() {
+        return inputs.angleMotorVoltage;
+    }
+
+    public Command setRollerPower(double power) {
+        return run(() -> io.setRollerMotorPower(power)).withName("set roller power");
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]:
```suggestion
        return runOnce(() -> io.setRollerMotorPower(power)).withName("set roller power");
```

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    public boolean hasNote() {
+        return inputs.hasNote;
+    }
+
+    public MutableMeasure<Voltage> getRollerMotorVoltage() {
+        return inputs.rollerMotorVoltage;
+    }
+
+    public MutableMeasure<Voltage> getAngleMotorVoltage() {
+        return inputs.angleMotorVoltage;
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Why are these needed?

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        INSTANCE = new Gripper(io);
+    }
+
+    public MutableMeasure<Angle> getCurrentAngle() {
+        return inputs.currentAngle;
+    }
+
+    public MutableMeasure<Angle> getAngleSetpoint() {
+        return inputs.angleSetpoint;
+    }
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: If these are needed to determine whether the gripper reached its setpoint, it's better to delete these and replace those with a `atSetpoint()` method.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Gripper extends SubsystemBase {
+    private static Gripper INSTANCE;
+    private final GripperIO io;
+    private final GripperInputsAutoLogged inputs = GripperIO.inputs;
+
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java:26]: You need to add an `@AutoLogOutput` for a `Pose3d` that represents the position of the gripper. This is so AdvantageScope will be able to visualize it. @vichik123 knows about that better than me.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 12 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java:26]: b16e92f2b9b1a6d1a7355b4c09d4e192cb0e74dd

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 10 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 51558eb55a5c6463524eb09b5b7d9fb59f610b22

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 9 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 77077cfe11feb411bc0d3ada7372a5d4db462838

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 10 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Not a relevant commit

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 11 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 4ea4e82a2d4135189590cc7751261f9eb86eccaa

Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: This is not an input. Inputs are data received from the IO, but checking whether the setpoint has been reached is "logic" code, and should be handled in the subsystem code.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 12 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java:26]: Not sure how to update it

Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java:26]: nevermind
bce75c91e6d13b897e3c760f78924f8356ca0989

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+
+    public Command setWristPosition(MutableMeasure<Angle> angle) {
+        return runOnce(() -> io.setAngle(angle)).withName("set wrist position");
+    }
+
+    @Override
+    public void periodic() {
+        pose3d =
+                new Pose3d(
+                        GripperConstants.GRIPPER_POSITION,
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Remember that the pose of the gripper depends on the height of the first stage of the elevator. So we need a way to pass this to the gripper.

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+    GripperInputsAutoLogged inputs = new GripperInputsAutoLogged();
+
+    void setRollerMotorPower(double power);
+
+    void setAngleMotorPower(double power);
+
+    void setAngle(MutableMeasure<Angle> angle);
+
+    void updateInputs(GripperInputs inputs);
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: No need to have the inputs as a parameter, it exists as a member in the implementations.

File: src/main/java/frc/robot/subsystems/gripper/GripperInputs.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+@AutoLog
+public class GripperInputs {
+    public MutableMeasure<Angle> currentAngle;
+    public MutableMeasure<Angle> angleSetpoint;
+    public double rollerPowerSetpoint;
+    public MutableMeasure<Voltage> angleMotorVoltage;
+    public MutableMeasure<Voltage> rollerMotorVoltage;
+    public boolean hasNote;
+    public MutableMeasure<Velocity<Velocity<Distance>>> acceleration;
+    public boolean atSetpoint;
```
Comment (vichik123) [inline src/main/java/frc/robot/subsystems/gripper/GripperInputs.java]: Like @katzuv said, this shouldn't be an input.

**Review by vichik123 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 11 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 165bed460a9ea16225582dd1cd7cd095bbec600e

File: src/main/java/frc/robot/subsystems/gripper/GripperIO.java (hunk 14 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperIO.java]: a554e0b5cf343ecb191494bf7cbf3e4f4c1c3394

File: src/main/java/frc/robot/subsystems/gripper/GripperInputs.java (hunk 15 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperInputs.java]: a554e0b5cf343ecb191494bf7cbf3e4f4c1c3394

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+    public static void initialize(GripperIO io) {
+        INSTANCE = new Gripper(io);
+    }
+
+    public boolean hasNote() {
+        return inputs.hasNote;
+    }
+
+    public boolean atSetpoint() {
+        return inputs.currentAngle.isNear(inputs.angleSetpoint, 0.02);
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 0.02 should be stored it a constant of type `Measure<Dimensionless>`.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Gripper extends SubsystemBase {
+    private static Gripper INSTANCE;
+    private final GripperIO io;
+    private final GripperInputsAutoLogged inputs = GripperIO.inputs;
+
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(0, 0);
+    @AutoLogOutput private Pose3d pose3d = new Pose3d(0, 0, 0, new Rotation3d());
```
Comment (katzuv) [inline src/main/java/frc/robot/subsystems/gripper/Gripper.java]: A variable name should rarely be identical to the class name. Try to think of a name that describes the variable's meaning, not type.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 16 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 0eb4bbbf1420f1f9b517fd4b1042369b6ed39f93

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 17 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: cc13c3af1c64574076d38581043e0144a088593d

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 13 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Like using elevator inputs in the gripper class?

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 16 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: This is a constant, it shouldn't be mutable.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 17 above)
Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Great!

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 13 above)
Comment (vichik123) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: Yeah actually that would work. Remember to call the static member in the `ElevatorIO` interface and not create a new set of inputs.

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 16 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: c7f81f291ce9f5b71be3890a6fdcb9fd61be6150

File: src/main/java/frc/robot/subsystems/gripper/GripperInputs.java (hunk 15 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/GripperInputs.java]: d66ef4afb7db3ba53292f6119a1ced03f66dd314

File: src/main/java/frc/robot/subsystems/gripper/Gripper.java (hunk 13 above)
Comment (GaiaZano05) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: 88691b89f6cb783836f0307304af20f246685063

Comment (katzuv) [reply src/main/java/frc/robot/subsystems/gripper/Gripper.java]: `elevatorInputs.currentHeight.mut_plus(...)` probably changes the elevator's inputs in the gripper's code which we don't want to happen. Instead of passing the whole inputs class, it's better to pass only a supplier for the elevator's height – classes should be [encapsulated](https://en.m.wikipedia.org/wiki/Encapsulation_(computer_programming)) and not export details which aren't needed. I would even make the inputs classes package-private to make sure that doesn't happen.
